### PR TITLE
[Feature] Support git archive in piperider compare

### PR DESCRIPTION
**Pull request by ctiml**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Feature

**What this PR does / why we need it**:
1. With git-archive, the untracked files are allowed
2. No need to switch branches

**Which issue(s) this PR fixes**:
sc-31292

**Special notes for your reviewer**:
for internal:
https://www.loom.com/share/9dda38174ed54818926dc7c65b4267f3

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

**Review by kentwelcome — COMMENTED** [review]
Due to our change to use `git archive` instead of using `git checkout`, can `piperider compare` support running under a dirty git repo?

File: piperider_cli/configuration.py [hunk 1]
```diff
@@ -325,18 +326,25 @@ def from_dbt_project(cls, dbt_project_path, dbt_profiles_dir=None):
 
         dbt_profile = dbtutil.load_dbt_profile(os.path.expanduser(dbt_profile_path))
 
+        console = Console()
         profile_name = dbt_project.get('profile', '')
-        target_name = dbt_profile.get(profile_name, {}).get('target', '')
+        profile_content = dbt_profile.get(profile_name, None)
+
+        if profile_content is None:
+            "Could not find profile named 'jaffle_shop'"
```
Comment (kentwelcome) [inline piperider_cli/configuration.py:334]: This line should be removed.

Comment (ctiml) [reply piperider_cli/configuration.py]: fixed

**Review by kentwelcome — APPROVED** [review]
LGTM
Just remember to remove the git repo dirty check logic before merging.

**Pull request merged**